src/Coordination/KeeperStorage.cpp

```diff
@@ -198,356 +176,650 @@ void KeeperStorage::Node::removeChild(StringRef child_path)
+                            session_id))
+                        onStorageInconsistency();
+
+                    return Coordination::Error::ZOK;
+                }
+                else if constexpr (std::same_as<DeltaType, KeeperStorage::UpdateNodeDelta>)
+                {
+                    auto node_it = container.find(path);
+                    if (node_it == container.end())
+                        onStorageInconsistency();
+
+                    if (operation.version != -1 && operation.version != node_it->value.stat.version)
+                        onStorageInconsistency();
+
+                    container.updateValue(path, operation.update_fn);
+                    return Coordination::Error::ZOK;
+                }
+                else if constexpr (std::same_as<DeltaType, KeeperStorage::RemoveNodeDelta>)
+                {
+                    if (!removeNode(path, operation.version))
+                        onStorageInconsistency();
+
+                    return Coordination::Error::ZOK;
+                }
+                else if constexpr (std::same_as<DeltaType, KeeperStorage::SetACLDelta>)
+                {
+                    auto node_it = container.find(path);
+                    if (node_it == container.end())
+                        onStorageInconsistency();
+
+                    if (operation.version != -1 && operation.version != node_it->value.stat.aversion)
+                        onStorageInconsistency();
+
+                    acl_map.removeUsage(node_it->value.acl_id);
+
+                    uint64_t acl_id = acl_map.convertACLs(operation.acls);
+                    acl_map.addUsage(acl_id);
+
+                    container.updateValue(path, [acl_id](KeeperStorage::Node & node) { node.acl_id = acl_id; });
+
+                    return Coordination::Error::ZOK;
+                }
+                else if constexpr (std::same_as<DeltaType, KeeperStorage::ErrorDelta>)
+                    return operation.error;
+                else if constexpr (std::same_as<DeltaType, KeeperStorage::SubDeltaEnd>)
+                {
+                    finish_subdelta = true;
+                    return Coordination::Error::ZOK;
+                }
+                else if constexpr (std::same_as<DeltaType, KeeperStorage::AddAuthDelta>)
+                {
+                    session_and_auth[operation.session_id].emplace_back(std::move(operation.auth_id));
+                    return Coordination::Error::ZOK;
+                }
+                else
+                {
+                    // shouldn't be called in any process functions
+                    onStorageInconsistency();
+                }
+            },
+            delta.operation);
+
+        if (result != Coordination::Error::ZOK)
+            return result;
+
+        if (finish_subdelta)
+            return Coordination::Error::ZOK;
+    }
+
+    return Coordination::Error::ZOK;
+}
+
+bool KeeperStorage::createNode(
```

Comment:
Slightly unclear what return value means. A node exists and we didn't create it? Or parent path didn't exist? Need a comment.